options.go

```diff
@@ -177,8 +197,38 @@ func IdentityResolverFunc(resolver IdentityResolver) Option {
 	}
 }
 
+// RingChecksumStatPeriod configures the period between emissions of the stat
+// 'ring.checksum-periodic'. Using a value <=0 (or RingChecksumStatPeriodNever)
+// will disable emission of this stat. Using a value in (0, 10ms) will return
+// an error, as that value is unrealistically small. Normal values must
+// therefore be >=10ms. RingChecksumStatPeriodDefault defines the default.
+func RingChecksumStatPeriod(period time.Duration) Option {
+	return func(r *Ringpop) error {
+		if period <= 0 {
+			period = RingChecksumStatPeriodNever
+		} else if period < 10*time.Millisecond {
+			return errors.New("RingChecksumStatPeriod invalid below 10 ms")
+		}
+		r.config.RingChecksumStatPeriod = period
+		return nil
+	}
+}
+
 // Default options
 
+// RingChecksumStatPeriodNever defines a "period" which disables
+// ring.checksum-periodic stat emission.
+const RingChecksumStatPeriodNever = time.Duration(-1)
```

Comment:
> Our code ought to use symbolic constants instead of magic numbers wherever possible; clients/users can do it as desired.

That makes sense. But it probably makes sense to define `RingChecksumStatPeriodNever` as the same thing we're telling users to use: `0`.

Otherwise we're testing with `time.Duration(-1)` but telling our users to use `0`.

I see the tests are testing the `<= 0` case though, which is a good thing. Nice one.